### PR TITLE
Nucleo L452RE board definition added

### PR DESCRIPTION
**Pull request by martinjaeger**
The board definitions are based on existing Nucleo L432KC and were tested with blinky app. In addition to that, I'm using a derived board definition with same MCU for my custom hardware.

### CODE REVIEW

File: boards/arm/nucleo_l452re/nucleo_l452re.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+identifier: nucleo_l452re
+name: NUCLEO-L452RE
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 64
```
Comment (erwango) [inline boards/arm/nucleo_l452re/nucleo_l452re.yaml]: 128

File: boards/arm/nucleo_l452re/nucleo_l452re.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+flash: 256
```
Comment (erwango) [inline boards/arm/nucleo_l452re/nucleo_l452re.yaml]: 512

File: boards/arm/nucleo_l452re/nucleo_l452re_defconfig [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+CONFIG_CLOCK_STM32_AHB_PRESCALER=1
+CONFIG_CLOCK_STM32_APB1_PRESCALER=1
+CONFIG_CLOCK_STM32_APB2_PRESCALER=1
+
+# console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+#enable pwm
+CONFIG_PWM=y
```
Comment (erwango) [inline boards/arm/nucleo_l452re/nucleo_l452re_defconfig]: PWM should not be enabled by default

File: boards/arm/nucleo_l452re/nucleo_l452re.dts [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/l4/stm32l452Xc.dtsi>
+
```
Comment (erwango) [inline boards/arm/nucleo_l452re/nucleo_l452re.dts]: Please add arduino header as .dtsi include

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by martinjaeger** [comment]
Thanks @erwango for the quick feedback. All requested changes should now be implemented.

I kept the copyright for untouched `arduino_r3_connector.dtsi` file and put my own copyright notice for files I made significant changes to. Tried to put only the SPDX Identifier for trivial files like the `Kconfig.board`, but zephyrbot complained about missing copyright... hope it's fine now.

Is it correct that the other warnings from zephyrbot can be ignored? I see that none of the other Nucleo boards has these bindings defined. And the line that's too long is an URL that should probably not be wrapped.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Comment by erwango** [comment]
@martinjaeger, there are still several comments which are not addressed, please check.
Also, you should not add a commit that fixes the comments, but directlty fix the initial commits.

**Comment by martinjaeger** [comment]
> @martinjaeger, there are still several comments which are not addressed, please check.
> Also, you should not add a commit that fixes the comments, but directlty fix the initial commits.

@erwango can you please be more specific which "several" comments are not addressed? I double-checked, but could not find any of your comments not being fixed.

Now also updated the initial commit instead of adding additional commits, as you requested.

BTW: What is the reason why PWM should not be enabled by default for this board, but it is enabled for e.g. nucleo_l476rg?

**Review by erwango — CHANGES_REQUESTED** [review]
> @erwango can you please be more specific which "several" comments are not addressed? I double-checked, but could not find any of your comments not being fixed.

When I performed the review yesterday, I was still able to see most of the point I commented (and github did not put the comments as "outdated"). They are indeed resolved now. 
One explication might be that I'm reviewing commit by commit, and since you didn't fixed comments within commits they were still present. I had no time to go more in detail.

> Now also amended the initial commit instead of adding additional commits, as you requested.

Ok thanks.

> BTW: What is the reason why PWM should not be enabled by default for this board, but it is enabled for e.g. nucleo_l476rg?

This is general board configuration guidelines. This has to be fixed in non complying boards (https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#default-board-configuration).

Last comment: please split the dts related changes in a separate commit.
And then we'll be good, thanks!

**Comment by martinjaeger** [comment]
> Last comment: please split the dts related changes in a separate commit.
> And then we'll be good, thanks!

Alright, here we go.

**Review by erwango — APPROVED** [review]

**Review by ioannisg — COMMENTED** [review]
Any reason for not enabling MPU @martinjaeger ?

**Comment by martinjaeger** [comment]
> Any reason for not enabling MPU @martinjaeger ?

No, just that it was not enabled by default in `nucleo_l432kc`. And it's also not mentioned in the porting guide linked above.

I suggest that we keep this as it is now and someone who is in charge of the STM boards unifies all board configurations with the correct default settings. For me it's a bit hard to guess which features should be enabled or not, if it's not consistent in the existing board definitions.

**Comment by erwango** [comment]
>>   Any reason for not enabling MPU @martinjaeger ?

> No, just that it was not enabled by default in nucleo_l432kc. And it's also not mentioned in the porting guide linked above.

Agreed. @ioannisg as discussed in another PR, we need to settle (/document if already settled) on guidelines on MPU activation. Either way I'm fine, but it has to be a documented.

**Comment by ioannisg** [comment]
> > > Any reason for not enabling MPU @martinjaeger ?
> 
> > No, just that it was not enabled by default in nucleo_l432kc. And it's also not mentioned in the porting guide linked above.
> 
> Agreed. @ioannisg as discussed in another PR, we need to settle (/document if already settled) on guidelines on MPU activation. Either way I'm fine, but it has to be a documented.

@martinjaeger @erwango it's fine with me if you want to get this PR merged as is and unify the policy of the MPU enabling in the STM32.

Right now, several other vendors (Nordic, NXP, etc.) have already a consistent behavior, i.e. enable ARM_MPU in the board, whenever MPU support is set in the SoC definition. For STM32 this seems to not be the case.

**Comment by erwango** [comment]
> Right now, several other vendors (Nordic, NXP, etc.) have already a consistent behavior, i.e. enable ARM_MPU in the board, whenever MPU support is set in the SoC definition. For STM32 this seems to not be the case.

Sure, I propose to get it documented and then convert non complying STM32 platforms.

**Review by ioannisg — APPROVED** [review]
Looks ok to me. I wonder @martinjaeger if you have addresses the 3 warnings listed by CI in this PR; looked fairly trivial.

**Comment by martinjaeger** [comment]
@ioannisg Re first two CI warnings: As far as I can see, none of the other Nucleo boards has these bindings defined, so I thought they might be ignored. And the comment line that's too long (3rd warning) is an URL that should probably not be wrapped.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**